### ISSUE TITLE
Add precompiled header support

### ISSUE THREAD

**Issue by Sipkab** [sakerbuild/saker.msvc#4]
Support creating and consuming precompiled headers with the build tasks.

**Comment by Sipkab** [comment]
According to the documentation of [/Yu](https://docs.microsoft.com/en-us/cpp/build/reference/yu-use-precompiled-header-file?view=vs-2019), the precompiled headers cannot be shared between machines.

Testing shows that the generated pch file includes paths of the included files and various other environment related information. This causes that when distributed compilation is used, the precompiled headers cannot be shared.

This means that the precompilation cannot be done separately of the source files in which it is included. 

The precompiled headers should be created on every build cluster where source files are compiled that use precompiled headers. This results in the precompiled headers being created on each cluster where a source file compilation is being dispatched to.

So to summarize, each cluster may need to perform the precompilation for a given precompiled header file. This can result in all clusters performing the precompilation.

**Comment by Sipkab** [comment]
Another thing that needs examination is how precompiled headers can be shared between different compilation passes. It could make sense, but may be a rarely occurring use-case.

For the current implementation we will not deal with this, but it is considered to be an enhancement in the future. This may need some build system improvements to properly implement shared data.
